testsuite/tests/typing-gadts/unexpected_existentials.ml

```diff
@@ -154,5 +154,5 @@ Line 1, characters 14-20:
 1 | class c = let Any _x = () in object end
                   ^^^^^^
 Error: Existential types are not allowed in bindings inside class definition,
-       but the constructor "Any" introduces existential types.
+       but this pattern introduces the existential type "$a".
```

Comment:
This changes is due to the code handling "unexpected" existentials at `typecore.ml` is looking for a existential which is not named "$Constr" before switching between
```ocaml
but the constructor "..." introduces existential types.
```
or 
```ocaml
but this pattern introduces the existential type "...".
```
This part of the error message needs to be simplified along this PR. I propose to keep only the constructor form.